packages/bot/src/scratch/index.js

```diff
@@ -80,218 +66,28 @@ export const scratchWorkspaceInit = async () => {
 
         window.addEventListener('resize', onWorkspaceResize);
         onWorkspaceResize();
-    } catch (error) {
-        // TODO: Handle error.
-        throw error;
-    }
-};
 
-const disableStrayBlocks = () => {
-    const top_blocks = Blockly.derivWorkspace.getTopBlocks();
+        const handleDragOver = e => {
+            e.stopPropagation();
+            e.preventDefault();
+            e.dataTransfer.dropEffect = 'copy'; // eslint-disable-line no-param-reassign
+        };
 
-    top_blocks.forEach(block => {
-        if (block.isMainBlock() || !block.isIndependentBlock()) {
-            block.setDisabled(true);
-        }
-    });
-};
+        const dropZone = document.body;
 
-export const loadWorkspace = xml => {
-    Blockly.Events.setGroup('load');
-    Blockly.mainWorkspace.clear();
+        dropZone.addEventListener('dragover', handleDragOver, false);
+        dropZone.addEventListener('drop', handleFileChange, false);
 
-    Blockly.Xml.domToWorkspace(xml, Blockly.mainWorkspace);
-    addLoadersFirst(xml).then(
-        () => {
-            Blockly.Events.setGroup(false);
-            globalObserver.emit('ui.log.success', translate('Blocks are loaded successfully'));
-        },
-        e => {
-            Blockly.Events.setGroup(false);
-            throw e;
-        }
-    );
-};
+        // resize run panel height
+        const run_drawer = document.getElementsByClassName('dc-drawer')[0];
+        const drawer_toggle = document.getElementsByClassName('dc-drawer__toggle')[0];
+        const drawer_height = `calc(100vh - ${   toolbar_heigh + 82  }px)`;
+        const drawer_top = `${toolbar_heigh + 2}px`;
 
-export const loadBlocks = (xml, dropEvent = {}) => {
-    const variables = xml.getElementsByTagName('variables');
-    if (variables.length > 0) {
-        Blockly.Xml.domToVariables(variables[0], Blockly.mainWorkspace);
+        run_drawer.style.height = drawer_toggle.style.height = drawer_height;
+        run_drawer.style.top = drawer_top;
+    } catch (error) {
+        // TODO: Handle error.
+        throw error;
     }
-    Blockly.Events.setGroup('load');
-    addLoadersFirst(xml).then(
-        loaders => {
-            const addedBlocks = [
-                ...loaders,
-                ...Array.from(xml.children)
-                    .map(block => addDomAsBlock(block))
-                    .filter(b => b),
-            ];
-            cleanUpOnLoad(addedBlocks, dropEvent);
-            globalObserver.emit('ui.log.success', translate('Blocks are loaded successfully'));
-        },
-        e => {
-            throw e;
-        }
-    );
 };
-
-export default class _Blockly {
```

Comment:
Please don't delete this class (yet). We still need `run()` and `stop()`.

Reply:
We're not using the class. You can always refer to binary repo for the `run()` and `stop()` logic